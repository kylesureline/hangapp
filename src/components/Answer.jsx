import React from 'react';
import { connect } from 'react-redux';

export const Answer = ({ word, type, player }) => {
  const guessedWord = player.guessedWord;
  const wordArray = word.split('');
  return (
    <div className="answer">
      <h2>{wordArray.map((letter, index) => {
        return wordArray[index] === guessedWord[index] ? (
          <span key={index}>{wordArray[index]}</span>
        ) : (
          <span key={index}>_</span>
        )
      })}</h2>
      <p>{!!type ? type : 'No word type available'}</p>
    </div>
  );
};
const mapStateToProps = (state) => ({
  word: state.player.answer.word,
  type: state.player.answer.type,
  player: state.player
});

export default connect(mapStateToProps)(Answer);
