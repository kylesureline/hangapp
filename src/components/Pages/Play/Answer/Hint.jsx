import React from "react";

export const Hint = ({ dictionary, answer, isOver }) => {
  const { showDefinition, showWordType } = dictionary;
  if (!answer.category) {
    // only the dictionary word object will not have a category set
    const { def, wordType } = answer;
    return (
      <h2 className="answer__hint">
        {(showDefinition || isOver) && (
          <span className="answer__hint__definition">{def}</span>
        )}
        {(showWordType || isOver) && (
          <span className="answer__hint__word-type">{wordType}</span>
        )}
      </h2>
    );
  } else if (answer.category === "recipe") {
    return (
      <h2 className="answer__hint">
        <span className="answer__hint__definition">
          {answer.tags.join(" ")}
        </span>
        <span className="answer__hint__word-type">{answer.category}</span>
      </h2>
    );
  } else if (answer.category === "dog") {
    return (
      <h2 className="answer__hint">
        <span className="answer__hint__definition">{answer.bredFor}</span>
        <span className="answer__hint__word-type">{answer.category}</span>
      </h2>
    );
  } else if (answer.category === "cat") {
    return (
      <h2 className="answer__hint">
        <span className="answer__hint__definition">{answer.temperament}</span>
        <span className="answer__hint__word-type">{answer.category}</span>
      </h2>
    );
  } else {
    return "";
  }
};
