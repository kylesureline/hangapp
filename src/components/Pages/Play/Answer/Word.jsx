import React from "react";

export const Word = ({ word }) => (
  <span className="word">
    {word.split("").map((letter, index) => (
      <span key={index} className="word__letter">
        {letter}
      </span>
    ))}
  </span>
);
