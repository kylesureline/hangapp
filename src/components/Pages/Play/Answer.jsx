import React from 'react';
import { Word } from '../../reusable/Word';
import { useSelector } from 'react-redux';
import { Hint } from './Hint';

export const Answer = () => {
  const { progress, answer, isOver } = useSelector(state => state.game);
  const { mode, words: wordsSettings } = useSelector(state => state.settings);

  return (
    <div className="answer">
      <h1 className="answer__value">
        {isOver ? (
          answer.words.map((word, index) => <Word key={index} word={word} />)
        ) : (
          progress.map((word, index) => <Word key={index} word={word} />)
        )}
      </h1>
      <Hint mode={mode} wordsSettings={wordsSettings} answer={answer} isOver={isOver} />
    </div>
  );
};
