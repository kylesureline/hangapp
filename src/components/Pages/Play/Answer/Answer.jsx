import React from "react";
import { Word } from "./Word";
import { useSelector } from "react-redux";
import { Hint } from "./Hint";

export const Answer = () => {
  const { progress, answer, isOver } = useSelector(state => state.game);
  const { dictionary } = useSelector(state => state.settings);

  return (
    <div className="answer">
      <h1 className="answer__value">
        {isOver
          ? answer.words.map((word, index) => <Word key={index} word={word} />)
          : progress.map((word, index) => <Word key={index} word={word} />)}
      </h1>
      <Hint dictionary={dictionary} answer={answer} isOver={isOver} />
    </div>
  );
};
