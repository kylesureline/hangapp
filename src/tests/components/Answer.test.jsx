import React from 'react';
import { shallow } from 'enzyme';
import { Answer } from '../../components/Answer.jsx';

test('should render Answer correctly', () => {
  const wrapper = shallow(
    <Answer
      word={'jazz'}
      type={'noun'}
      player={{
        guessedWord: ['j']
      }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
