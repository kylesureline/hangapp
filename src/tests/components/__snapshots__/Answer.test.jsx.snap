// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Answer correctly 1`] = `
<div
  className="answer"
>
  <h2>
    <span
      key="0"
    >
      j
    </span>
    <span
      key="1"
    >
      _
    </span>
    <span
      key="2"
    >
      _
    </span>
    <span
      key="3"
    >
      _
    </span>
  </h2>
  <p>
    noun
  </p>
</div>
`;
